implementations/rust/ockam/ockam_node/src/storage/database/migrations/sqlx_migration.rs

```diff
@@ -37,7 +43,68 @@ impl NodesMigration {
         NodeNameIdentityAttributes::migrate_attributes_node_name(pool).await?;
         AuthorityAttributes::migrate_authority_attributes_to_members(pool).await?;
         PolicyTrustContextId::migrate_update_policies(pool).await?;
+        SplitPolicies::migrate_policies(pool).await?;
+        Ok(())
+    }
+
+    #[cfg(test)]
+    pub(crate) async fn migrate_schema_before(
+        &self,
+        pool: &SqlitePool,
+        version: i64,
+    ) -> Result<()> {
+        let migrator = sqlx::migrate!("./src/storage/database/migrations");
+        let mut conn = pool.acquire().await.into_core()?;
+        let mut transaction = conn.begin().await.into_core()?;
+        for m in migrator.migrations.iter() {
+            if m.version >= version {
+                break;
+            }
+            transaction.execute(&*m.sql).await.into_core()?;
+        }
+        transaction.commit().await.into_core()?;
+        Ok(())
+    }
+
+    #[cfg(test)]
+    pub(crate) async fn migrate_schema_single(
```

Comment:
Is the result the same as calling `migrate_schema_before`?

Reply:
No, this runs only one migration. This is used once you have the old data in place and you want to apply the migration you want to test. [Here is](https://github.com/build-trust/ockam/blob/d8fe696ad1e1491b70018ffe81ad0574b7a42f95/implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_20231231100000_node_name_identity_attributes.rs#L90) an example of that.

Reply:
@adrianbenavides yes, good example. So if in that example you replace `migrate_schema_single` call with `migrate_schema_before` call, will the result be the same? If yes, do we really need both?

Reply:
That will fail because we can't run the same migration more than once with these functions. They run the migrations manually, and not through the `Migrator`, so they are not added to the `_sqlx_migrations` table.